src/M6Web/Component/RedisMock/RedisMock.php

```diff
@@ -988,10 +988,13 @@ public function zadd($key, ...$args) {
 
         $isNew = !isset(self::$dataValues[$this->storage][$key][$member]);
 
-        self::$dataValues[$this->storage][$key][$member] = (int) $score;
-        self::$dataTypes[$this->storage][$key]     = 'zset';
-        if (array_key_exists($key, self::$dataTtl[$this->storage]))
-        {
+        if (!is_int($score) && !is_float($score)) {
+            throw new \InvalidArgumentException('Score should be either an integer or a float.');
+        }
```

Comment:
What about using [`is_numeric`](https://www.php.net/manual/en/function.is-numeric.php)? 🤔 
There is a trick in comment to convert a numeric string into actual int/float (`$var + 0`).
Maybe something like this:
```php
if (!is_numeric($score)) {
     throw new \InvalidArgumentException('Score should be either an integer or a float.');
}
$score += 0; // convert potential string value to int or float
```